### ISSUE TITLE
'method_missing': undefined method `this'

### ISSUE THREAD

**Issue by johnbeynon** [heroku/heroku#1905]
Brought this across from StackOverflow (http://stackoverflow.com/questions/35548753/heroku-server-restart-command-throwing-error)

`heroku restart` is throwing:

```
lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:2159:in `method_missing': undefined method `this' for #<Gem::Specification:0x3fd6490129e4 heroku-3.42.15> (NoMethodError)
from /Users/Topform/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/specification.rb:1057:in `find_active_stub_by_path'
from /Users/Topform/.rvm/rubies/ruby-2.1.4/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:64:in `require'
from /Users/Topform/.rvm/gems/ruby-2.1.4@skinnymint/gems/heroku-3.42.15/bin/heroku:11:in `<top (required)>'
from /Users/Topform/.rvm/gems/ruby-2.1.4@skinnymint/bin/heroku:23:in `load'
from /Users/Topform/.rvm/gems/ruby-2.1.4@skinnymint/bin/heroku:23:in `<main>'
from /Users/Topform/.rvm/gems/ruby-2.1.4@skinnymint/bin/ruby_executable_hooks:15:in `eval'
from /Users/Topform/.rvm/gems/ruby-2.1.4@skinnymint/bin/ruby_executable_hooks:15:in `<main>'
```

seems to be when trying to include the updater :
https://github.com/heroku/heroku/blob/master/bin/heroku#L11

But from the StackTrace it looks like it's a Gem'ed install.

**Comment by samar-mishra** [comment]
Yes I am facing this issue !

**Comment by ransombriggs** [comment]
Could you please run `gem update heroku` and try again?  I have shipped a fix for this error in version `3.42.37` of the gem.
